### PR TITLE
logging: fix ProgressTracer bug

### PR DESCRIPTION
**Pull request by patrickkenney9801**
Fixes a small bug when spans are closed by developers via the tracer
Makes assembler script compatible with normal output as well as tracer output

### CODE REVIEW

File: libsupport/src/ProgressTracer.cpp [hunk 1]
```diff
@@ -24,10 +24,11 @@ katana::ProgressTracer::StartActiveSpan(
 void
 katana::ProgressTracer::FinishActiveSpan() {
   if (active_span_ != nullptr) {
-    if (!active_span_->IsFinished()) {
-      active_span_->Finish();
+    auto old_active_span = active_span_;
+    active_span_ = old_active_span->GetParentSpan();
+    if (!old_active_span->IsFinished()) {
+      old_active_span->Finish();
```
Comment (arekay) [inline libsupport/src/ProgressTracer.cpp:30]: I don't have a lot of context, but is it possible to be in a much deeper span-nest, so this should really be a `while old_active_span->GetParentSpan()` construct?

File: scripts/assemble_trace.py [hunk 2]
```diff
@@ -217,6 +217,8 @@ def assemble_traces(args):
 
     stream = len(args.log) == 0
     for line in fileinput.input(args.log[0] if not stream else ("-",)):
+        if line[0] == '"':
```
Comment (arekay) [inline scripts/assemble_trace.py:220]: Is a `"` expected? Comments on why would be useful.

**Review by arekay — APPROVED** [review]
LGTM

File: scripts/assemble_trace.py (hunk 2 above)
Comment (patrickkenney9801) [reply scripts/assemble_trace.py:220]: It breaks the python JSON parser for some reason and it occurs on exactly 1 line in the import logs

File: libsupport/src/ProgressTracer.cpp (hunk 1 above)
Comment (patrickkenney9801) [reply libsupport/src/ProgressTracer.cpp:30]: good catch, added some logic to handle this recursively in case someone is ending spans early

**Pull request merged**